sdk/devcenter/Azure.Developer.DevCenter/src/Generated/DevBoxesClient.cs

```diff
@@ -222,24 +223,26 @@ public virtual Response GetScheduleByPool(string poolName, string scheduleName,
         /// </item>
         /// </list>
         /// </summary>
+        /// <param name="projectName"> The DevCenter Project upon which to execute operations. </param>
         /// <param name="userId"> The AAD object id of the user. If value is 'me', the identity is taken from the authentication context. The default value is "me". </param>
         /// <param name="devBoxName"> The name of a Dev Box. </param>
         /// <param name="context"> The request context, which can override default behaviors of the client pipeline on a per-call basis. </param>
-        /// <exception cref="ArgumentNullException"> <paramref name="userId"/> or <paramref name="devBoxName"/> is null. </exception>
-        /// <exception cref="ArgumentException"> <paramref name="userId"/> or <paramref name="devBoxName"/> is an empty string, and was expected to be non-empty. </exception>
+        /// <exception cref="ArgumentNullException"> <paramref name="projectName"/>, <paramref name="userId"/> or <paramref name="devBoxName"/> is null. </exception>
+        /// <exception cref="ArgumentException"> <paramref name="projectName"/>, <paramref name="userId"/> or <paramref name="devBoxName"/> is an empty string, and was expected to be non-empty. </exception>
         /// <exception cref="RequestFailedException"> Service returned a non-success status code. </exception>
         /// <returns> The response returned from the service. </returns>
-        /// <include file="Docs/DevBoxesClient.xml" path="doc/members/member[@name='GetDevBoxByUserAsync(string,string,RequestContext)']/*" />
-        public virtual async Task<Response> GetDevBoxByUserAsync(string userId, string devBoxName, RequestContext context)
+        /// <include file="Docs/DevBoxesClient.xml" path="doc/members/member[@name='GetDevBoxAsync(string,string,string,RequestContext)']/*" />
+        public virtual async Task<Response> GetDevBoxAsync(string projectName, string userId, string devBoxName, RequestContext context)
```

Comment:
I do agree with the suggestion but that's not how the current version of the generator handles required parameters. Here's the summary of the current behavior: https://github.com/Azure/autorest.csharp/issues/3475#issuecomment-1579947570

In short, default values for required parameters are ignored. A possible workaround is to mark this parameter as optional in the transform but that wouldn't work because it's a path parameter (hence required by definition).

Manual updates to this parameter in the SDK are pretty much useless since they will be overridden by the next code generation (which happens often and outside my control).

Reply:
Resolving this thread for now.

Reply:
@olegoid what about this: https://github.com/Azure/autorest/tree/main/docs/extensions#x-ms-client-default